imod/msw/sprinkling.py

```diff
@@ -51,23 +55,31 @@ class Sprinkling(MetaSwapPackage):
         "trajectory",
     )
 
+    _regrid_method = SprinklingRegridMethod()
+
     def __init__(
         self,
         max_abstraction_groundwater: xr.DataArray,
         max_abstraction_surfacewater: xr.DataArray,
-        well: Mf6Wel,
```

Comment:
I cant find in the code base where this object is instantiated except for some test classes. So i don't know how feasible the comment below is

Would it be possible to introduce a view or wrapper around the original Well object?
When the well object is replaced outside this class the view/wrapper needs to be updated
The advantage is that you can pass the wrapper to the init method which avoids extending the write and render with arguments that other packages don't need but are now forced to accept

```
@dataclass
class Mf6WelView
  well : Mf6Wel
```

Reply:
The MODFLOW6 simulation also only generates ``Mf6Wel`` objects during the call to ``write``.

The disadvantage of giving a Mf6Wel to the init is that it complicates the ``Sprinkling.regrid_like`` operation, as extra logic has to be introduced to regrid Mf6Wel objects.  

Furthermore the ``Sprinkling.dump`` method, which we want to implement, would require extra logic to separate the Well attribute from the Sprinkling class, separately store the well in a NetCDF, and load the Well NetCDF again and attaching it to ``Sprinkling.well`` in the ``Sprinkling.from_file`` constructor.